torchtitan/datasets/hf_datasets.py

```diff
@@ -10,7 +10,14 @@
 import torch
 from torch.distributed.checkpoint.stateful import Stateful
 from torch.utils.data import IterableDataset
-from torchdata.stateful_dataloader import StatefulDataLoader
+
+try:
+    from torchdata.stateful_dataloader import StatefulDataLoader
+except ModuleNotFoundError as e:
```

Comment:
Nit: maybe also include import error here incase user did not install torch data at all? For my case I had tried to install but the version was wrong.

Reply:
It turns out `ModuleNotFoundError` is a subclass of `ImportError`, so changing to `ImportError` here.